apis/rds/v1alpha1/custom.go

```diff
@@ -0,0 +1,17 @@
+package v1alpha1
+
+import "github.com/crossplane-contrib/terrajet/pkg/terraform/resource"
+
+var DBClusterExternalNamer = resource.ExternalNamer{
+	SelfVarPath: "github.com/crossplane-contrib/provider-tf-aws/apis/rds/v1alpha1.DBClusterExternalNamer",
```

Comment:
Could we get rid of `SelfVarPath` by appending custom configuration (`DBClusterExternalNamer` here) in the `init()` function to a global map? I believe we already have the package information (`github.com/crossplane-contrib/provider-tf-aws/apis/rds/v1alpha1`) for a given resource while generating terraformed functions. This could also help to access per resource. Or we could enforce a naming convention `RDSClusterExternalNamer` and check if that exists?

e.g. 
```
var externalNamers map[string]resource.ExternalNamer
```

```
c.ExternalNamer = externalNamers[name]
```

Reply:
Synced with @turkenh offline and we'll go with his proposed design in his next PR.

Reply:
I am working on this and trying to define a custom configuration API that will include External Name config, field reference config, and also possible future configurations (e.g. which fields should go into connection details secret). One inconsistency I realized, we need most of the custom configurations during code generation (e.g. which fields to omit for external name, for which fields we should add reference fields etc.), however, there are also some that we would need at runtime, like calling external name configure function and reference resolve extractor. We could make it work for external name configure func by passing an instance of custom configuration to controllers (down to terraformed), however, it seems not to be possible for reference extractor since for terrajet we need to pass the full path of the function to crossplane-tools.

Reply:
Yes, we are able to work around the disparity of some of the fields being needed in generator and some in provider for external name but for references, we have to give a string and I couldn't find a way to know the package and name of a function object in runtime using reflection. It seems like for references, we have to give a string.